src/IO/GeoJSONWriter.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Brick\Geo\IO;
+
+use Brick\Geo\Exception\GeometryIOException;
+use Brick\Geo\Geometry;
+use Brick\Geo\GeometryCollection;
+use Brick\Geo\LineString;
+use Brick\Geo\MultiLineString;
+use Brick\Geo\MultiPoint;
+use Brick\Geo\MultiPolygon;
+use Brick\Geo\Point;
+use Brick\Geo\Polygon;
+
+/**
+ * Converter class from Geometry to GeoJSON.
+ */
+class GeoJSONWriter
+{
+    /**
+     * @param Geometry $geometry The geometry to export as GeoJSON.
+     *
+     * @return string The GeoJSON representation of the given geometry.
+     *
+     * @throws GeometryIOException If the given geometry cannot be exported as GeoJSON.
+     */
+    public function write(Geometry $geometry) : string
+    {
+        if ($geometry instanceof GeometryCollection
+            && ! is_subclass_of($geometry, GeometryCollection::class)) {
+            return $this->writeFeatureCollection($geometry);
+        }
+
+        return $this->genGeoJSONString($this->formatGeoJSONGeometry($geometry));
+    }
+
+    /**
+     * @param Geometry $geometry
+     *
+     * @return array
+     *
+     * @throws GeometryIOException
+     */
+    private function formatGeoJSONGeometry(Geometry $geometry) : array
+    {
+        if ($geometry instanceof Point) {
+            $type = 'Point';
+        } elseif ($geometry instanceof MultiPoint) {
+            $type = 'MultiPoint';
+        } elseif ($geometry instanceof LineString) {
+            $type = 'LineString';
+        } elseif ($geometry instanceof MultiLineString) {
+            $type = 'MultiLineString';
+        } elseif ($geometry instanceof Polygon) {
+            $type = 'Polygon';
+        } elseif ($geometry instanceof MultiPolygon) {
+            $type = 'MultiPolygon';
+        } else {
+            throw GeometryIOException::unsupportedGeometryType($geometry->geometryType());
+        }
+
+        return [
+            'type' => $type,
+            'coordinates' => $geometry->toArray()
+        ];
+    }
+
+    /**
+     * @param GeometryCollection $geometryCollection
+     *
+     * @return string
+     * @throws GeometryIOException
+     */
+    private function writeFeatureCollection(GeometryCollection $geometryCollection) : string
+    {
+        $geojsonArray = [
+            'type' => 'FeatureCollection',
+            'features' => []
+        ];
+
+        if ($geometryCollection->isEmpty()) {
+            return $this->genGeoJSONString($geojsonArray);
+        }
```

Comment:
Is this empty check required? AFAICS, the foreach() will just do nothing in this case, so this code block is unnecessary.

Reply:
Just me trying to be a bit more paranoid than needed

Resolved in Commit: a8f07f1